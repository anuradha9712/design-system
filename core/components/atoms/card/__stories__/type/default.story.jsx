import * as React from 'react';
import { Card } from '@/index';

// CSF format story
export const defaultCard = () => {
  return (
    <Card>
      <div className="p-8" />
    </Card>
  );
};

export default {
  title: 'Layout/Card/Type/Default Card',
  component: Card,
};
