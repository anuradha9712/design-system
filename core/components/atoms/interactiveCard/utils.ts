export const removeItem = (targetItem: any, arr = []) => {
  return arr.filter((item: any) => item !== targetItem);
};
