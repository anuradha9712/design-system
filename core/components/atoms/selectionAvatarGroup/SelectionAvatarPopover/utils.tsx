import React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<Element | undefined>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  listRef?: any,
  withSearch?: boolean
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'Enter':
      handleEnterKey(focusedOption);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    default:
      break;
  }
};

const handleEnterKey = (focusedOption: Element | undefined) => {
  (focusedOption as HTMLElement)?.click();
};

const navigateSelectedOption = (
  index: number,
  withSearch?: boolean,
  listItems?: any,
  direction?: string,
  listRef?: any,
  setFocusedOption?: React.Dispatch<React.SetStateAction<Element | undefined>>
) => {
  console.log('tttindexxx', index);
  // if (withSearch) {
  //   const searchInput = listRef.current.querySelector('[data-test="DesignSystem-SelectionAvatar--Input"]');
  //   console.log('ttt inside iff');
  //   if ((index === 0 && direction === 'up') || (index === listItems.length - 2 && direction === 'down')) {
  //     console.log('tttfocus search');
  //     searchInput.focus();
  //     setFocusedOption && setFocusedOption(searchInput);
  //     return;
  //   }
  // }
  const searchInput = listRef.current.querySelector('[data-test="DesignSystem-SelectionAvatar--Input"]');
  searchInput.focus();
  setFocusedOption && setFocusedOption(searchInput);
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<Element | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  const listItems = listRef.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  console.log('tttwithSearchwithSearch', withSearch, 'index', index, 'listItems[index]', listItems[index]);

  if (index === -1) {
    console.log('ttt11');
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else if (
    (withSearch && index === 0 && direction === 'up') ||
    (withSearch && index === listItems.length - 1 && direction === 'down')
  ) {
    console.log('ttt22');

    navigateSelectedOption(index, withSearch, listItems, direction, listRef, setFocusedOption);
    // return;
  } else {
    console.log('ttt333');

    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;

    const targetOption = listItems[index];

    (targetOption as HTMLElement).focus();
    setFocusedOption && setFocusedOption(targetOption);
    targetOption.scrollIntoView({ block: 'center' });
  }
};

export const handleInputKeyDown = (
  event: React.KeyboardEvent,
  listRef: any,
  setFocusedOption: React.Dispatch<React.SetStateAction<Element | undefined>>
) => {
  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      targetOption = listItems[listItems.length - 1];
      break;
    case 'ArrowDown':
      event.preventDefault();
      targetOption = listItems[0];
      break;
    default:
      break;
  }

  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
