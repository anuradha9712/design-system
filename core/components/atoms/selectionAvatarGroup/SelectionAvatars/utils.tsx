import React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>
) => {
  switch (event.key) {
    case 'Enter':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowDown':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowUp':
      setOpenPopover?.(true);
      setHighlightLastItem?.(true);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<Element | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  if (withSearch && position === 'down') {
    const searchInput = listRef.current?.querySelectorAll('[data-test="DesignSystem-SelectionAvatar--Input"]');
    console.log('aa22', searchInput);
    searchInput[0]?.focus();
    setFocusedOption && setFocusedOption(searchInput);
    return;
  }

  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  console.log('listReflistRef', listRef.current, 'listItems', listItems);
  let targetOption;

  if (position === 'down') {
    targetOption = listItems[0];
  } else {
    targetOption = listItems[listItems.length - 1];
  }
  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
