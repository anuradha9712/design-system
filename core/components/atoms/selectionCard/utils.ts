export const removeItem = (targetItem: string, arr: string[]) => {
  return arr.filter((item: string) => item !== targetItem);
};

export const handleMultiSelect = (
  selectedOptions: string[],
  isItemSelected: boolean,
  setSelectedOptions: React.Dispatch<React.SetStateAction<string[]>>,
  id: string
) => {
  let resultList = [...selectedOptions];
  if (isItemSelected) {
    resultList = removeItem(id, selectedOptions);
  } else {
    resultList.push(id);
  }
  setSelectedOptions(resultList);
};

export const handleSingleSelect = (
  selectedOptions: string[],
  isItemSelected: boolean,
  setSelectedOptions: React.Dispatch<React.SetStateAction<string[]>>,
  id: string
) => {
  let resultList = [...selectedOptions];
  if (isItemSelected) {
    resultList = [];
  } else {
    resultList = [id];
  }

  setSelectedOptions(resultList);
};
