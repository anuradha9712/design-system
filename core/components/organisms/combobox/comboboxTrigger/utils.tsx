import React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  // setFocusedOption?: React.Dispatch<React.SetStateAction<Element | undefined>>,
  setHighlightFirstItem?: any,
  setHighlightLastItem?: any
) => {
  console.log('handle key down e.key', event.key);

  switch (event.key) {
    case 'ArrowUp':
      setOpenPopover?.(true);
      setHighlightLastItem(true);
      // focusListItem('up', setFocusedOption);
      // requestAnimationFrame(() => focusListItem('up', setFocusedOption));
      break;
    case 'ArrowDown':
      setOpenPopover?.(true);
      setHighlightFirstItem(true);

      // requestAnimationFrame(() => focusListItem('down', setFocusedOption));
      break;
    case 'Escape':
      setOpenPopover?.(false);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<Element | undefined>>
) => {
  const listItems = document.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;
  // if (position === 'down') {
  //   targetOption = listItems[0];
  //   (listItems[0] as HTMLElement)?.focus();
  // } else {
  //   (listItems[listItems.length - 1] as HTMLElement)?.focus();
  // }

  if (position === 'down') {
    console.log('rrr down', listItems);
    targetOption = listItems[0];
  } else {
    targetOption = listItems[listItems.length - 1];
  }
  console.log('rrrtargetOption', targetOption);
  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
