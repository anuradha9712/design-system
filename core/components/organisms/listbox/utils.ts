const focusOption = (targetIndex: number, totalLength: number, direction: string) => {
  let startIndex = targetIndex;
  const endIndex = direction === 'down' ? totalLength : -1;

  while (startIndex !== endIndex) {
    const selectedElement = document.querySelector(`[data-list-item-index="${startIndex}"]`) as HTMLElement;

    if (selectedElement && selectedElement.getAttribute('data-disabled') !== 'true') {
      selectedElement && selectedElement.focus();
      break;
    }

    if (direction === 'down') {
      startIndex++;
    } else {
      startIndex--;
    }
  }
};

export const onKeyDown = (event: React.KeyboardEvent, totalLength: number) => {
  const sourceElement = event.target as HTMLElement;
  const sourceIndex = sourceElement.getAttribute('data-list-item-index');
  const targetIndex = (sourceIndex && parseInt(sourceIndex, 10)) || 0;

  switch (event.key) {
    case 'ArrowDown':
      event.preventDefault();
      focusOption(targetIndex + 1, totalLength, 'down');
      break;
    case 'ArrowUp':
      event.preventDefault();
      focusOption(targetIndex - 1, totalLength, 'up');
      break;
    case 'Enter': {
      event.preventDefault();
      sourceElement.click();
      break;
    }
    default:
      break;
  }
};
