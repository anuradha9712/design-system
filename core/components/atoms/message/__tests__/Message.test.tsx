import * as React from 'react';
import { render } from '@testing-library/react';
import { Message, Text } from '@/index';
import { MessageProps as Props } from '@/index.type';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';

const appearances: Props['appearance'][] = ['alert', 'info', 'warning', 'success'];
const title = 'Title goes here';
const description = 'Description goes here';
const sizes: Props['size'][] = ['small', 'regular'];
const actions = (
  <>
    <Text className="mr-5 cursor-pointer" appearance="link">
      Action 1
    </Text>
    <Text className="cursor-pointer" appearance="link">
      Action 2
    </Text>
  </>
);

describe('Message component', () => {
  const mapper: Record<string, any> = {
    appearance: valueHelper(appearances, { required: true, iterate: true }),
    title: valueHelper(title, { required: true }),
    description: valueHelper(description, { required: true }),
    actions: valueHelper(actions, { required: true }),
    size: valueHelper(sizes, { iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<Message {...attr}>Description goes here</Message>);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Message component', () => {
  it('renders description', () => {
    const { getByTestId } = render(<Message appearance="info" description={description} />);
    expect(getByTestId('DesignSystem-Message--Description').textContent).toMatch(description);
  });
});

describe('Message Component with overwrite class', () => {
  it('overwrite Message class', () => {
    const { getByTestId } = render(<Message className="MessageClass" description={description} />);
    expect(getByTestId('DesignSystem-Message')).toHaveClass('MessageClass');
  });
});

describe('Message component with prop:Title', () => {
  it('renders title', () => {
    const { getByTestId } = render(<Message appearance="info" title={'Masala'} description={description} />);
    expect(getByTestId('DesignSystem-Message--Title').textContent).toMatch('Masala');
    expect(getByTestId('DesignSystem-Message--Icon')).toHaveClass('Message-icon--withTitle');
  });

  it('should not have Message-title class if title is not present ', () => {
    const { getByTestId, queryByTestId } = render(<Message appearance="info" description={description} />);
    expect(queryByTestId('DesignSystem-Message--Title')).not.toBeInTheDocument();
    expect(getByTestId('DesignSystem-Message--Icon')).not.toHaveClass('Message-icon--withTitle');
  });
});

describe('Message component with prop:appearance', () => {
  appearances.forEach((appearance) => {
    it(`should have the Message--${appearance} class when appearance = ${appearance}`, () => {
      const { getByTestId } = render(<Message appearance={appearance} description={description} />);
      expect(getByTestId('DesignSystem-Message')).toHaveClass(`Message--${appearance}`);
    });
  });
});

describe('Message component with prop: actions', () => {
  it('renders actions', () => {
    const { getByTestId } = render(<Message appearance="info" description={description} actions={actions} />);
    expect(getByTestId('DesignSystem-Message--actions')).toBeInTheDocument();
  });
});

describe('Message component with prop:size', () => {
  it('applies small size classes and icon size when size = small', () => {
    const { getByTestId } = render(
      <Message appearance="info" title={title} description={description} size="small" actions={actions} />
    );
    expect(getByTestId('DesignSystem-Message')).toHaveClass('Message--small');

    const iconEl = getByTestId('DesignSystem-Message--Icon');
    expect(iconEl).toHaveClass('Message-icon--small');
    expect(iconEl).toHaveStyle('width: 14px;');

    expect(getByTestId('DesignSystem-Message--Title')).toHaveClass('Message-heading--small');
    expect(getByTestId('DesignSystem-Message--Description')).toHaveClass('Message-text--small');
    expect(getByTestId('DesignSystem-Message--actions')).toHaveClass('Message-actions--small');
  });

  it('applies regular size classes explicitly when size = regular', () => {
    const { getByTestId } = render(
      <Message appearance="info" title={title} description={description} size="regular" actions={actions} />
    );
    expect(getByTestId('DesignSystem-Message')).not.toHaveClass('Message--small');

    const iconEl = getByTestId('DesignSystem-Message--Icon');
    expect(iconEl).toHaveClass('Message-icon--regular');
    expect(iconEl).toHaveStyle('width: 16px;');

    expect(getByTestId('DesignSystem-Message--Title')).toHaveClass('Message-heading--regular');
    expect(getByTestId('DesignSystem-Message--Description')).toHaveClass('Message-text--regular');
    expect(getByTestId('DesignSystem-Message--actions')).toHaveClass('Message-actions--regular');
  });

  it('uses regular size by default when no size prop provided', () => {
    const { getByTestId } = render(
      <Message appearance="info" title={title} description={description} actions={actions} />
    );

    expect(getByTestId('DesignSystem-Message')).not.toHaveClass('Message--small');

    const iconEl = getByTestId('DesignSystem-Message--Icon');
    expect(iconEl).toHaveClass('Message-icon--regular');
    expect(iconEl).not.toHaveClass('Message-icon--small');
    expect(iconEl).toHaveStyle('width: 16px;');

    expect(getByTestId('DesignSystem-Message--Title')).toHaveClass('Message-heading--regular');
    expect(getByTestId('DesignSystem-Message--Title')).not.toHaveClass('Message-heading--small');
    expect(getByTestId('DesignSystem-Message--Description')).toHaveClass('Message-text--regular');
    expect(getByTestId('DesignSystem-Message--Description')).not.toHaveClass('Message-text--small');
    expect(getByTestId('DesignSystem-Message--actions')).toHaveClass('Message-actions--regular');
  });

  it('applies correct size classes without title', () => {
    const { getByTestId } = render(
      <Message appearance="info" description={description} size="small" actions={actions} />
    );

    expect(getByTestId('DesignSystem-Message')).toHaveClass('Message--small');

    const iconEl = getByTestId('DesignSystem-Message--Icon');
    expect(iconEl).toHaveClass('Message-icon--small');
    expect(iconEl).not.toHaveClass('Message-icon--withTitle');
    expect(iconEl).toHaveStyle('width: 14px;');

    expect(getByTestId('DesignSystem-Message--Description')).toHaveClass('Message-text--small');
    expect(getByTestId('DesignSystem-Message--actions')).toHaveClass('Message-actions--small');
  });

  it('applies size classes correctly with children instead of description', () => {
    const { getByTestId } = render(
      <Message appearance="info" title={title} size="small" actions={actions}>
        <div>Custom child content</div>
      </Message>
    );

    expect(getByTestId('DesignSystem-Message')).toHaveClass('Message--small');
    expect(getByTestId('DesignSystem-Message--Icon')).toHaveClass('Message-icon--small');
    expect(getByTestId('DesignSystem-Message--Title')).toHaveClass('Message-heading--small');
    expect(getByTestId('DesignSystem-Message--actions')).toHaveClass('Message-actions--small');
  });

  sizes.forEach((size) => {
    it(`should have correct icon size and classes for ${size} size`, () => {
      const expectedIconSize = size === 'small' ? 14 : 16;
      const { getByTestId } = render(
        <Message appearance="info" title={title} description={description} size={size} actions={actions} />
      );

      const iconEl = getByTestId('DesignSystem-Message--Icon');
      expect(iconEl).toHaveStyle(`width: ${expectedIconSize}px;`);
      expect(iconEl).toHaveClass(`Message-icon--${size}`);
      expect(getByTestId('DesignSystem-Message--Title')).toHaveClass(`Message-heading--${size}`);
      expect(getByTestId('DesignSystem-Message--Description')).toHaveClass(`Message-text--${size}`);
      expect(getByTestId('DesignSystem-Message--actions')).toHaveClass(`Message-actions--${size}`);
    });
  });
});
