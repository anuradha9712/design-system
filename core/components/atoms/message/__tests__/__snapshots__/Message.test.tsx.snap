// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component 
  appearance: "alert", title: "Title goes here", description: "Description goes here", actions: "[ReactNode]"
 1`] = `
<body>
  <div>
    <div
      class="Message Message--alert"
      data-test="DesignSystem-Message"
    >
      <i
        class="material-symbols material-symbols-rounded Icon Icon--alert Message-icon--alert Message-icon--regular Message-icon--withTitle"
        data-test="DesignSystem-Message--Icon"
        role="button"
        style="font-size: 16px; width: 16px;"
      >
        error
      </i>
      <div>
        <h5
          class="Heading Heading--s Heading--default Message-heading--alert Message-heading--regular"
          data-test="DesignSystem-Message--Title"
        >
          Title goes here
        </h5>
        <span
          class="Text Text--default Text--regular Message-text--alert Message-text--regular"
          data-test="DesignSystem-Message--Description"
        >
          Description goes here
        </span>
        <div
          class="Message-actions Message-actions--regular"
          data-test="DesignSystem-Message--actions"
        >
          <span
            class="Text Text--link Text--regular mr-5 cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 1
          </span>
          <span
            class="Text Text--link Text--regular cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 2
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Message component 
  appearance: "alert", title: "Title goes here", description: "Description goes here", actions: "[ReactNode]", size: "regular"
 1`] = `
<body>
  <div>
    <div
      class="Message Message--alert"
      data-test="DesignSystem-Message"
    >
      <i
        class="material-symbols material-symbols-rounded Icon Icon--alert Message-icon--alert Message-icon--regular Message-icon--withTitle"
        data-test="DesignSystem-Message--Icon"
        role="button"
        style="font-size: 16px; width: 16px;"
      >
        error
      </i>
      <div>
        <h5
          class="Heading Heading--s Heading--default Message-heading--alert Message-heading--regular"
          data-test="DesignSystem-Message--Title"
        >
          Title goes here
        </h5>
        <span
          class="Text Text--default Text--regular Message-text--alert Message-text--regular"
          data-test="DesignSystem-Message--Description"
        >
          Description goes here
        </span>
        <div
          class="Message-actions Message-actions--regular"
          data-test="DesignSystem-Message--actions"
        >
          <span
            class="Text Text--link Text--regular mr-5 cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 1
          </span>
          <span
            class="Text Text--link Text--regular cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 2
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Message component 
  appearance: "alert", title: "Title goes here", description: "Description goes here", actions: "[ReactNode]", size: "small"
 1`] = `
<body>
  <div>
    <div
      class="Message Message--alert Message--small"
      data-test="DesignSystem-Message"
    >
      <i
        class="material-symbols material-symbols-rounded Icon Icon--alert Message-icon--alert Message-icon--small Message-icon--withTitle"
        data-test="DesignSystem-Message--Icon"
        role="button"
        style="font-size: 14px; width: 14px;"
      >
        error
      </i>
      <div>
        <h5
          class="Heading Heading--s Heading--default Message-heading--alert Message-heading--small"
          data-test="DesignSystem-Message--Title"
        >
          Title goes here
        </h5>
        <span
          class="Text Text--default Text--regular Message-text--alert Message-text--small"
          data-test="DesignSystem-Message--Description"
        >
          Description goes here
        </span>
        <div
          class="Message-actions Message-actions--small"
          data-test="DesignSystem-Message--actions"
        >
          <span
            class="Text Text--link Text--regular mr-5 cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 1
          </span>
          <span
            class="Text Text--link Text--regular cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 2
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Message component 
  appearance: "info", title: "Title goes here", description: "Description goes here", actions: "[ReactNode]"
 1`] = `
<body>
  <div>
    <div
      class="Message Message--info"
      data-test="DesignSystem-Message"
    >
      <i
        class="material-symbols material-symbols-rounded Icon Icon--info Message-icon--info Message-icon--regular Message-icon--withTitle"
        data-test="DesignSystem-Message--Icon"
        role="button"
        style="font-size: 16px; width: 16px;"
      >
        info
      </i>
      <div>
        <h5
          class="Heading Heading--s Heading--default Message-heading--info Message-heading--regular"
          data-test="DesignSystem-Message--Title"
        >
          Title goes here
        </h5>
        <span
          class="Text Text--default Text--regular Message-text--info Message-text--regular"
          data-test="DesignSystem-Message--Description"
        >
          Description goes here
        </span>
        <div
          class="Message-actions Message-actions--regular"
          data-test="DesignSystem-Message--actions"
        >
          <span
            class="Text Text--link Text--regular mr-5 cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 1
          </span>
          <span
            class="Text Text--link Text--regular cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 2
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Message component 
  appearance: "info", title: "Title goes here", description: "Description goes here", actions: "[ReactNode]", size: "regular"
 1`] = `
<body>
  <div>
    <div
      class="Message Message--info"
      data-test="DesignSystem-Message"
    >
      <i
        class="material-symbols material-symbols-rounded Icon Icon--info Message-icon--info Message-icon--regular Message-icon--withTitle"
        data-test="DesignSystem-Message--Icon"
        role="button"
        style="font-size: 16px; width: 16px;"
      >
        info
      </i>
      <div>
        <h5
          class="Heading Heading--s Heading--default Message-heading--info Message-heading--regular"
          data-test="DesignSystem-Message--Title"
        >
          Title goes here
        </h5>
        <span
          class="Text Text--default Text--regular Message-text--info Message-text--regular"
          data-test="DesignSystem-Message--Description"
        >
          Description goes here
        </span>
        <div
          class="Message-actions Message-actions--regular"
          data-test="DesignSystem-Message--actions"
        >
          <span
            class="Text Text--link Text--regular mr-5 cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 1
          </span>
          <span
            class="Text Text--link Text--regular cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 2
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Message component 
  appearance: "info", title: "Title goes here", description: "Description goes here", actions: "[ReactNode]", size: "small"
 1`] = `
<body>
  <div>
    <div
      class="Message Message--info Message--small"
      data-test="DesignSystem-Message"
    >
      <i
        class="material-symbols material-symbols-rounded Icon Icon--info Message-icon--info Message-icon--small Message-icon--withTitle"
        data-test="DesignSystem-Message--Icon"
        role="button"
        style="font-size: 14px; width: 14px;"
      >
        info
      </i>
      <div>
        <h5
          class="Heading Heading--s Heading--default Message-heading--info Message-heading--small"
          data-test="DesignSystem-Message--Title"
        >
          Title goes here
        </h5>
        <span
          class="Text Text--default Text--regular Message-text--info Message-text--small"
          data-test="DesignSystem-Message--Description"
        >
          Description goes here
        </span>
        <div
          class="Message-actions Message-actions--small"
          data-test="DesignSystem-Message--actions"
        >
          <span
            class="Text Text--link Text--regular mr-5 cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 1
          </span>
          <span
            class="Text Text--link Text--regular cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 2
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Message component 
  appearance: "success", title: "Title goes here", description: "Description goes here", actions: "[ReactNode]"
 1`] = `
<body>
  <div>
    <div
      class="Message Message--success"
      data-test="DesignSystem-Message"
    >
      <i
        class="material-symbols material-symbols-rounded Icon Icon--success Message-icon--success Message-icon--regular Message-icon--withTitle"
        data-test="DesignSystem-Message--Icon"
        role="button"
        style="font-size: 16px; width: 16px;"
      >
        check_circle
      </i>
      <div>
        <h5
          class="Heading Heading--s Heading--default Message-heading--success Message-heading--regular"
          data-test="DesignSystem-Message--Title"
        >
          Title goes here
        </h5>
        <span
          class="Text Text--default Text--regular Message-text--success Message-text--regular"
          data-test="DesignSystem-Message--Description"
        >
          Description goes here
        </span>
        <div
          class="Message-actions Message-actions--regular"
          data-test="DesignSystem-Message--actions"
        >
          <span
            class="Text Text--link Text--regular mr-5 cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 1
          </span>
          <span
            class="Text Text--link Text--regular cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 2
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Message component 
  appearance: "success", title: "Title goes here", description: "Description goes here", actions: "[ReactNode]", size: "regular"
 1`] = `
<body>
  <div>
    <div
      class="Message Message--success"
      data-test="DesignSystem-Message"
    >
      <i
        class="material-symbols material-symbols-rounded Icon Icon--success Message-icon--success Message-icon--regular Message-icon--withTitle"
        data-test="DesignSystem-Message--Icon"
        role="button"
        style="font-size: 16px; width: 16px;"
      >
        check_circle
      </i>
      <div>
        <h5
          class="Heading Heading--s Heading--default Message-heading--success Message-heading--regular"
          data-test="DesignSystem-Message--Title"
        >
          Title goes here
        </h5>
        <span
          class="Text Text--default Text--regular Message-text--success Message-text--regular"
          data-test="DesignSystem-Message--Description"
        >
          Description goes here
        </span>
        <div
          class="Message-actions Message-actions--regular"
          data-test="DesignSystem-Message--actions"
        >
          <span
            class="Text Text--link Text--regular mr-5 cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 1
          </span>
          <span
            class="Text Text--link Text--regular cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 2
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Message component 
  appearance: "success", title: "Title goes here", description: "Description goes here", actions: "[ReactNode]", size: "small"
 1`] = `
<body>
  <div>
    <div
      class="Message Message--success Message--small"
      data-test="DesignSystem-Message"
    >
      <i
        class="material-symbols material-symbols-rounded Icon Icon--success Message-icon--success Message-icon--small Message-icon--withTitle"
        data-test="DesignSystem-Message--Icon"
        role="button"
        style="font-size: 14px; width: 14px;"
      >
        check_circle
      </i>
      <div>
        <h5
          class="Heading Heading--s Heading--default Message-heading--success Message-heading--small"
          data-test="DesignSystem-Message--Title"
        >
          Title goes here
        </h5>
        <span
          class="Text Text--default Text--regular Message-text--success Message-text--small"
          data-test="DesignSystem-Message--Description"
        >
          Description goes here
        </span>
        <div
          class="Message-actions Message-actions--small"
          data-test="DesignSystem-Message--actions"
        >
          <span
            class="Text Text--link Text--regular mr-5 cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 1
          </span>
          <span
            class="Text Text--link Text--regular cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 2
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Message component 
  appearance: "warning", title: "Title goes here", description: "Description goes here", actions: "[ReactNode]"
 1`] = `
<body>
  <div>
    <div
      class="Message Message--warning"
      data-test="DesignSystem-Message"
    >
      <i
        class="material-symbols material-symbols-rounded Icon Icon--warning Message-icon--warning Message-icon--regular Message-icon--withTitle"
        data-test="DesignSystem-Message--Icon"
        role="button"
        style="font-size: 16px; width: 16px;"
      >
        warning
      </i>
      <div>
        <h5
          class="Heading Heading--s Heading--default Message-heading--warning Message-heading--regular"
          data-test="DesignSystem-Message--Title"
        >
          Title goes here
        </h5>
        <span
          class="Text Text--default Text--regular Message-text--warning Message-text--regular"
          data-test="DesignSystem-Message--Description"
        >
          Description goes here
        </span>
        <div
          class="Message-actions Message-actions--regular"
          data-test="DesignSystem-Message--actions"
        >
          <span
            class="Text Text--link Text--regular mr-5 cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 1
          </span>
          <span
            class="Text Text--link Text--regular cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 2
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Message component 
  appearance: "warning", title: "Title goes here", description: "Description goes here", actions: "[ReactNode]", size: "regular"
 1`] = `
<body>
  <div>
    <div
      class="Message Message--warning"
      data-test="DesignSystem-Message"
    >
      <i
        class="material-symbols material-symbols-rounded Icon Icon--warning Message-icon--warning Message-icon--regular Message-icon--withTitle"
        data-test="DesignSystem-Message--Icon"
        role="button"
        style="font-size: 16px; width: 16px;"
      >
        warning
      </i>
      <div>
        <h5
          class="Heading Heading--s Heading--default Message-heading--warning Message-heading--regular"
          data-test="DesignSystem-Message--Title"
        >
          Title goes here
        </h5>
        <span
          class="Text Text--default Text--regular Message-text--warning Message-text--regular"
          data-test="DesignSystem-Message--Description"
        >
          Description goes here
        </span>
        <div
          class="Message-actions Message-actions--regular"
          data-test="DesignSystem-Message--actions"
        >
          <span
            class="Text Text--link Text--regular mr-5 cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 1
          </span>
          <span
            class="Text Text--link Text--regular cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 2
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Message component 
  appearance: "warning", title: "Title goes here", description: "Description goes here", actions: "[ReactNode]", size: "small"
 1`] = `
<body>
  <div>
    <div
      class="Message Message--warning Message--small"
      data-test="DesignSystem-Message"
    >
      <i
        class="material-symbols material-symbols-rounded Icon Icon--warning Message-icon--warning Message-icon--small Message-icon--withTitle"
        data-test="DesignSystem-Message--Icon"
        role="button"
        style="font-size: 14px; width: 14px;"
      >
        warning
      </i>
      <div>
        <h5
          class="Heading Heading--s Heading--default Message-heading--warning Message-heading--small"
          data-test="DesignSystem-Message--Title"
        >
          Title goes here
        </h5>
        <span
          class="Text Text--default Text--regular Message-text--warning Message-text--small"
          data-test="DesignSystem-Message--Description"
        >
          Description goes here
        </span>
        <div
          class="Message-actions Message-actions--small"
          data-test="DesignSystem-Message--actions"
        >
          <span
            class="Text Text--link Text--regular mr-5 cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 1
          </span>
          <span
            class="Text Text--link Text--regular cursor-pointer"
            data-test="DesignSystem-Text"
          >
            Action 2
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;
